src/solvers/strings/string_refinement.cpp

```diff
@@ -1047,8 +1002,7 @@ static optionalt<exprt> get_array(
     return nil_exprt();
   }
 
-  const exprt arr_val =
-    simplify_expr(adjust_if_recursive(super_get(arr), ns), ns);
+  const exprt arr_val = simplify_expr(super_get(arr), ns);
```

Comment:
How do we know this does the right thing now? Would all string tests start failing otherwise?

Reply:
Yes essentially. `simplify_expr` fails an assertion if there are some type mismatches.